pkg/ebpf/c/tracee.bpf.c

```diff
@@ -2111,6 +2169,50 @@ static __always_inline unsigned short get_inode_mode_from_fd(u64 fd)
     return READ_KERN(f_inode->i_mode);
 }
 
+static __always_inline void fetch_syscalls_addresses(event_data_t *data){
+    int key = 0;
+    u64 *table_ptr = bpf_map_lookup_elem(&syscalls_to_check_map, (void *)&key);
+    if (table_ptr == NULL){
+        return ;
+    }
+
+    unsigned long *syscall_table_addr = (unsigned long*) *table_ptr;
+    u64 idx;
+    u64* sys_num;                  // pointer to syscall_number
+    u64 snum;                       // syscall_number
+    unsigned long saddr = 0;        // syscall address
+    int number_of_hooked_syscalls = 0;
+    #if defined(bpf_target_x86)
+        number_of_hooked_syscalls = NUMBER_OF_SYSCALLS_X86;
+        u64 syscall_address[NUMBER_OF_SYSCALLS_X86];
+    #elif defined(bpf_target_arm64)
+        number_of_hooked_syscalls = NUMBER_OF_SYSCALLS_ARM;
+        u64 syscall_address[NUMBER_OF_SYSCALLS_ARM];
+    #else
+
+        return
+    #endif
+
+    __builtin_memset(syscall_address, 0, sizeof(syscall_address));
+    // the map should look like [syscall_table][syscall number 1][syscall number 2][syscall number 3]...
+    #pragma unroll
+    for(int i =0; i < number_of_hooked_syscalls; i++){
+        idx = i+1;
```

Comment:
yep, the PR was Alon's loaded kallsyms_map that you merged already.
BTW, those last 3 comments are because I didn't delete the old function, the new `invoke_print_syscall_table_event ` has all the fixes (: